astropy/utils/data_info.py

```diff
@@ -103,7 +104,7 @@ def dtype_info_name(dtype):
     return out
 
 
-def data_info_factory(names, funcs):
+def data_info_factory(names, funcs, apply_format=False):
```

Comment:
Let's make ``apply_format=True`` the default!

Reply:
This code is used to enable two options for `info()`, which are `'attributes'` and `'stats'`. The default option is `'attributes'`, which prints the `dtype`, `unit`, `format` and `description` attributes of the columns. It doesn't make much sense to try to apply the column format to those attributes, so it is not clear that setting `apply_format=True` by default would be useful.

Furthermore, setting the default value to `True` might cause problems to anyone who might be using this factory to enable more options.